### PR TITLE
Issue #358 put on ss state

### PR DESCRIPTION
**Pull request by konstan**
Extended generic PUT on RTP with state transition from Ready to Provisioning in case the RT is ss:state.  Prerequisites: run should be `mutable`, current state should be `Ready`, requested state should be `Provisioning`.

Connected to #358

### CODE REVIEW

File: jar-service/src/main/java/com/sixsq/slipstream/run/RuntimeParameterResource.java [hunk 1]
```diff
@@ -218,4 +262,14 @@ private void logTimeDiff(String msg, long before, long after) {
 	protected void logTimeDiff(String msg, long before) {
 		logTimeDiff(msg, before, System.currentTimeMillis());
 	}
+
+	private boolean isGlobalStateParameter() { return RuntimeParameter.GLOBAL_STATE_KEY.equals(key); }
```
Comment (schaubl) [inline jar-service/src/main/java/com/sixsq/slipstream/run/RuntimeParameterResource.java]: I think a new line is welcome here :)

Comment (konstan) [inline jar-service/src/main/java/com/sixsq/slipstream/run/RuntimeParameterResource.java]: I recall IntelliJ separated, but I recall reverting it for some reason :-/.  I'm separating onto different lines now.  Thanks.

File: jar-service/src/main/java/com/sixsq/slipstream/run/RuntimeParameterResource.java [hunk 2]
```diff
@@ -145,6 +145,9 @@ public void update(Representation entity) throws ResourceException,
 				abortOrReset(value, em);
 				setValue(value);
 			}
+		} else if (isGlobalStateParameter()) {
+			States newState = attemptChangeGlobalState(value);
+			setValue(newState.toString());
```
Comment (schaubl) [inline jar-service/src/main/java/com/sixsq/slipstream/run/RuntimeParameterResource.java]: With this change, we cannot force the State anymore.
The ability to force the state already helped us in the past to unblock some particular Run for some customers.
So I would prefer if we can keep this ability. Maybe by adding the following code between the line 190 and 191:
```java
} else if (getUser().isSuper()) {
    newState = toNewState;
```

File: jar-service/pom.xml [hunk 3]
```diff
@@ -200,8 +200,8 @@
 				<groupId>org.apache.maven.plugins</groupId>
 				<artifactId>maven-compiler-plugin</artifactId>
 				<configuration>
-					<target>1.6</target>
-					<source>1.6</source>
+					<target>1.7</target>
+					<source>1.7</source>
```
Comment (schaubl) [inline jar-service/pom.xml]: It should be updated to 1.7 everywhere or defined on the top level pom if possible.
Out of curiosity, what did you do which require 1.7 ?

File: jar-service/src/main/java/com/sixsq/slipstream/run/RuntimeParameterResource.java [hunk 4]
```diff
@@ -169,23 +172,64 @@ private String extractValueFromEntity(Representation entity) {
...
+	}
+
 	@Post
-	public void completeCurrentNodeState(Representation entity) {
+	public void completeCurrentNodeStateOrChangeGlobalState(Representation entity) {
 
 		String nodeName = runtimeParameter.getNodeName();
-		States newState = null;
-
+		States newState;
```
Comment (schaubl) [inline jar-service/src/main/java/com/sixsq/slipstream/run/RuntimeParameterResource.java]: Does this change come from an hint of Intellij ? 
I would suggest the follwing:
```java
States newState = attemptCompleteCurrentNodeState(nodeName);
```

File: jar-service/pom.xml (hunk 3 above)
Comment (konstan) [inline jar-service/pom.xml]: I initially used multiple-catch statements, but then when I was testing and running the server in Eclipse (yes ... still in Eclipse :)), this part of the code was failing in the JVM, so I reverted back to separate catches.  Here is the corresponding issue to change it everywhere #365

Comment (schaubl) [inline jar-service/pom.xml]: Ok. I did a grep into the whole code and I only found 
```xml
<source>1.6</source>
```
in `SlipStreamServer/jar-service/pom.xml` 
and in `SlipStreamServer/static-content/target/results/META-INF/maven/org.clojure/clojure/pom.xml`

Comment (konstan) [inline jar-service/pom.xml]: #365 wasn't worked on yet.  After fixing it, the source/target versions should be defined in the parent pom as properties.

File: jar-service/src/main/java/com/sixsq/slipstream/run/RuntimeParameterResource.java (hunk 2 above)
Comment (konstan) [inline jar-service/src/main/java/com/sixsq/slipstream/run/RuntimeParameterResource.java]: Agreed.  Adding.

File: jar-service/src/main/java/com/sixsq/slipstream/run/RuntimeParameterResource.java (hunk 4 above)
Comment (konstan) [inline jar-service/src/main/java/com/sixsq/slipstream/run/RuntimeParameterResource.java]: Overlooked to update it after a refactor.  Updating.  Thanks.

**Pull request merged**